test/box/info.test.lua

```diff
@@ -17,6 +17,8 @@ box.info.status
 string.len(box.info.uptime) > 0
 string.match(box.info.uptime, '^[1-9][0-9]*$') ~= nil
 box.info.cluster.uuid == box.space._schema:get{'cluster'}[2]
+type(box.info.hostname) == 'string'
```

Comment:
After this `os.execute("hostname") ~= ''` to make sure that real hostname is not an empty string.

Reply:
It seems like `os.execute()` doesn't work this way:
```
tarantool> os.execute('hostname') == 'gkashkin-vps'
gkashkin-vps
---
- false
...

tarantool> os.execute('hostname')
gkashkin-vps
---
- 0
...

tarantool> os.execute('hostname') == 0
gkashkin-vps
---
- true
...
```

Reply:
(And I check that it is not empty string the line below.)

Reply:
Sorry, I mean `io.popen` or `popen.shell`. Execute `hostname` via popen, obtain value via stdout and compare it with value obtained using `box.info.hostname`.

Reply:
Sure, will do.

Reply:
Added [another test](https://github.com/tarantool/tarantool/blob/ce73fa2f5697f2017b45730434b10bbf34d11aa3/test/box-luatest/gh_8605_add_box_info_hostname_test.lua) for that.

Reply:
Thanks!

Reply:
Well, there is a problem with hostname on [redos](https://github.com/tarantool/tarantool/actions/runs/5030912165/jobs/9023648604?pr=8651):
```
[036] sh: hostname: command not found
[036] #   expected: {exit_code = 0, state = "exited"}
[036] #   actual: {exit_code = 127, state = "exited"}
```

Reply:
I suspect there is no hostname command there. Let's skip the test if it is so (either by the exit code or use 'type/command/which hostname' beforehand).

Reply:
The exit code 127 is [standartized](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_08_02).

Reply:
Added `skip_if()` to the test.

Reply:
No `:close()` at skipping the test, but this is negligible due to popen object's GC hook.